packages/block-directory/src/store/controls.js

```diff
@@ -1,153 +1,71 @@
- * @param {Function} onLoad  Callback function on success.
- * @param {Function} onError Callback function on failure.
- */
-const loadScript = ( asset, onLoad, onError ) => {
+export const loadScript = ( asset ) => {
 	if ( ! asset ) {
-		return;
+		return Promise.reject( new Error( 'No script found.' ) );
 	}
-	const existing = document.querySelector( `script[src="${ asset.src }"]` );
-	if ( existing ) {
-		existing.parentNode.removeChild( existing );
-	}
-	const script = document.createElement( 'script' );
-	script.src = typeof asset === 'string' ? asset : asset.src;
-	script.onload = onLoad;
-	script.onerror = onError;
-	document.body.appendChild( script );
+	return new Promise( ( resolve, reject ) => {
+		const existing = document.querySelector( `script[src="${ asset }"]` );
+		if ( existing ) {
+			existing.parentNode.removeChild( existing );
+		}
+		const script = document.createElement( 'script' );
+		script.src = asset;
+		script.onload = () => resolve( true );
+		script.onerror = () => reject( new Error( 'Error loading script.' ) );
+		document.body.appendChild( script );
+	} );
 };
 
 /**
- * Loads CSS file.
+ * Loads a CSS file.
  *
- * @param {*} asset the url for the CSS file.
+ * @param {string} asset The url for this file.
+ *
+ * @return {Promise} Promise which will resolve when the asset is added.
  */
-const loadStyle = ( asset ) => {
+export const loadStyle = ( asset ) => {
 	if ( ! asset ) {
-		return;
+		return Promise.reject( new Error( 'No style found.' ) );
 	}
-	const link = document.createElement( 'link' );
-	link.rel = 'stylesheet';
-	link.href = typeof asset === 'string' ? asset : asset.src;
-	document.body.appendChild( link );
+	return new Promise( ( resolve, reject ) => {
+		const link = document.createElement( 'link' );
+		link.rel = 'stylesheet';
+		link.href = asset;
+		link.onload = () => resolve( true );
+		link.onerror = () => reject( new Error( 'Error loading script.' ) );
+		document.body.appendChild( link );
+	} );
 };
 
 /**
  * Load the asset files for a block
  *
- * @param {Array} assets A collection of URL for the assets.
+ * @param {Array} assets A collection of URLs for the assets.
  *
  * @return {Object} Control descriptor.
  */
-export function* loadAssets( assets ) {
+export function loadAssets( assets ) {
 	return {
 		type: 'LOAD_ASSETS',
 		assets,
 	};
 }
 
 const controls = {
-	SELECT: createRegistryControl(
-		( registry ) => ( { storeName, selectorName, args } ) => {
-			return registry.select( storeName )[ selectorName ]( ...args );
-		}
-	),
-	DISPATCH: createRegistryControl(
-		( registry ) => ( { storeName, dispatcherName, args } ) => {
-			return registry.dispatch( storeName )[ dispatcherName ]( ...args );
-		}
-	),
-	API_FETCH( { request } ) {
-		return wpApiFetch( { ...request } );
-	},
 	LOAD_ASSETS( { assets } ) {
-		return new Promise( ( resolve, reject ) => {
-			if ( Array.isArray( assets ) ) {
-				let scriptsCount = 0;
+		const scripts = assets.map( ( asset ) =>
```

Comment:
With the API being premature, and the control of block.json files being in the author's hand, its possible that we get non .js/.css files.

Reply:
That's fair— do you think we should do this validation here in the loading code, or back in the API itself?

Reply:
actually, rethinking this— on the basis of "escaping late", I'll add a check in loadScript/loadStyle to verify that at least the extension is valid.

If there ends up being a valid reason to load another file type, we can change this conditional then.

Reply:
Cool.